Thumbnail/ConsumerThumbnail.php

```diff
@@ -23,16 +24,20 @@ class ConsumerThumbnail implements ThumbnailInterface
 
     protected $backend;
 
+    protected $dispatcher;
+
     /**
-     * @param string                                              $id
-     * @param ThumbnailInterface                                  $thumbnail
-     * @param \Sonata\NotificationBundle\Backend\BackendInterface $backend
+     * @param string                   $id
+     * @param ThumbnailInterface       $thumbnail
+     * @param BackendInterface         $backend
+     * @param EventDispatcherInterface $dispatcher
      */
-    public function __construct($id, ThumbnailInterface $thumbnail, BackendInterface $backend)
+    public function __construct($id, ThumbnailInterface $thumbnail, BackendInterface $backend, EventDispatcherInterface $dispatcher = null)
     {
-        $this->id        = $id;
-        $this->thumbnail = $thumbnail;
-        $this->backend   = $backend;
+        $this->id           = $id;
+        $this->thumbnail    = $thumbnail;
+        $this->backend      = $backend;
+        $this->dispatcher   = $dispatcher;
```

Comment:
@Soullivaneuh I'll wait on that. If @rande thinks that the EventDispatcher argument should be optional, there won't be a need for a deprecation message.